content/articles/object-based-image-analysis-using-matlab/index.md

```diff
@@ -0,0 +1,200 @@
+
+### Introduction
+Object-based image analysis is the processing of an image based on the classification of its pixels to get more helpful information on the image. For example, such data can be based on height, edges, and boundaries. Matlab provides an interactive environment for object-based image analysis by executing functions used in analysis or inbuilt apps for image processing.
+
+Images contain objects with distinct regions. These regions have features such as area, perimeter, shapes, and height. Matlab allows for analysis of these properties using image analyzer functions or region props to obtain data from these images.
+
+Object-based image analysis is useful, especially in analyzing satellite maps, machine vision, and obtaining information based on object characteristics in an image.
+
+### Table of contents
+- [Prerequisites](#prerequisites)
+- [Edge detection](#edge-detection)
+- [Methods of edge detection](#methods-of-edge-detection)
+- [Object boundary detection](#object-boundary-detection)
+- [Boundary detection using morphological operations](#boundary-detection-using-morphological-operations)
+- [Obtaining boundary by erosion](#obtaining-boundary-by-erosion)
+- [Obtaining boundary by dilation](#obtaining-boundary-by-dilation)
+- [Highlighting text in an image](#highlighting-text-in-an-image)
+- [Labeling objects in images](#labeling-objects-in-images)
+- [Conclusion](#conclusion)
+
+### Prerequisites
+To follow along with this tutorial, you'll need:
+- [MATLAB](https://www.mathworks.com/products/get-matlab.html?s_tid=gn_getml) installed.
+- Basic understanding of [MATLAB](https://www.section.io/engineering-education/getting-started-with-matlab/) basics.
+
+
+### Edge detection 
+Edge detection is the identification of points within an image. These points are where the image has excellent contrast, and they are the defining points of the image. Edge detection work by detecting changes in brightness of the image pixels.
+
+Edge detection is useful in image segmentation and data extraction for comparison, objects separation, computer vision, and machine learning. Matlab supports a variety of functions that aids in edge detection.
+
+### Methods of edge detection
+There are several methods used in edge detection in images. These methods are used with the primary function `edge`.  We will demonstrate most of these methods using a similar image and then compare the outcome to find a more suitable way to provide a more detailed output.
+
+All these methods only accept a grayscale image input; hence it is essential to fast convert the RGB image to a grayscale image, conversion of RGB to gray is done using the function `rgb2gray`. The initial stage of all image processes is importing the image to the Matlab workspace. Importation of image is done using the function `imread('imagefolderpath')`. 
+
```

Comment:
Ensure all functions have opening and closing brackets, e.g `rgb2gray()`. Edit where applicable throughout the article!